pennylane/ops/qubit/non_parametric_ops.py

```diff
@@ -53,9 +52,9 @@ def num_params(self):
     def label(self, decimals=None, base_label=None):
         return base_label or "H"
 
-    @classmethod
-    def _matrix(cls, *params):
-        return cls.matrix
+    @staticmethod
+    def _matrix(*params):
+        return np.array([[INV_SQRT2, INV_SQRT2], [INV_SQRT2, -INV_SQRT2]])
```

Comment:
Here and below: matrix is not defined in a variable called by a classmethod, but is defined in the static method. If necessary, subclasses can use classmethods to have `cls` available, but it shouldn't be encouraged so that we get as little dependency on what is available in the class as possible...